### PR TITLE
Add maintenance and decommissioning stages

### LINKED ISSUE #252: Add specific fields for project stage start and end dates
SIF's SOURCE platform includes the following fields for start and end dates for project stages:

* Project_Definition_Start_Date
* Project_Definition_End_Date
* Project_Preparation_End_Date
* Procurement_End_Date
* Construction_End_Date
* O&M_End_Date

These stages mostly seem to map to the stages in the CoST IDS:

SOURCE stage | SOURCE description | IDS stage | IDS description
-- | -- | -- | --
Definition | The description of the strategic context in which the project lies as well as the identification its key characteristics and stakeholders (ends with the decision to mobilise resource for project appraisal) | Identification | the decision to develop a project within the budget and programme of a project owner.
Preparation | TBC | Preparation | The feasibility study, environmental and social impact assessment, general scoping of the project, establishing the packaging and procurement strategy, preliminary statutory requirements on environmental and land impacts, and the resulting budget authorization.
Procurement | TBC | Procurement | The procurement of the planning, design and works according to the procurement strategy.
Construction | TBC | Implementation | The implementation of the planning, design and works according to the procurement strategy.
Operation & Maintenance | TBC | n/a | n/a

**Proposal**

Add the following `Period` objects to the OC4IDS schema:

path | name | description
-- | -- | --
`identificationPeriod` | Identification Period | Identification of key project characteristics and stakeholders, and the decision to develop the project within the budget and programme of the project owner. Also known as project definition.
`preparationPeriod` | Preparation Period | The feasibility study, environmental and social impact assessment, general scoping of the project, establishing the packaging and procurement strategy, preliminary statutory requirements on environmental and land impacts, and the resulting budget authorization.
`procurementPeriod` | Procurement Period | The procurement of the planning, design and works according to the procurement strategy.
`implementationPeriod` | Implementation Period | The implementation of the planning, design and works according to the procurement strategy.
`operationAndMaintenancePeriod` | Operation and Maintenance Period | The technical operation, maintenance and/or provision of services relating to the infrastructure asset

Alternatively, we could add a flexible `milestones` array at the project level and use `Milestone.Code` to identify the end of each stage. However, given the close alignment between SOURCE and the CoST IDS, I think explicit fields should be OK and will be easier for users.

### LINKED ISSUE #391: Update project status codelist and getting started documentation to incorporate maintenance and decommissioning
The latest review of OC4IDS has revealed the need to identifying additional stages related to the end of an infrastructure project, names operation and decommissioning. In addition this review has highlighted some inconsistencies with how existing stages are sometimes referred to:

- implementation is sometimes referred to as construction
- procurement is sometimes referred to as tender management

In the case of the additional codes the International Coalition for Sustainable Infrastructure’s [Infrastructure Pathways](https://infrastructure-pathways.org/) and the [RICS ICMS](https://www.rics.org/content/dam/ricsglobal/documents/standards/icms-3-explained.pdf) provide descriptions that can be adapted for use in OC4IDS:

<div dir="ltr" style="margin-left:0pt;" align="left" id="docs-internal-guid-0371dd13-7fff-dcb4-e522-5c6f7e506e5c">

Stage | Infrastructure Pathways | RICS ICMS
-- | -- | --
Operation | Operations and Maintenance: In this phase, infrastructure owners and operators, including governments, have an opportunity to incorporate climate resilience into the use of infrastructure, through rethinking traditional approaches to inspections, maintenance, day-to-day operations and emergency response to tackle climate change impacts. There are opportunities to influence the resilience of both infrastructure that was originally designed with resilience in mind and infrastructure that was not, including retrofit strategies to enhance resilience. | Operation costs are costs incurred or carbon emitted in running andmanaging a constructed asset, including administrative support services, insurances, energy and environmental/ regulatory inspection costs. Operational costs include operator’s site overheads and taxes and levies, though these are excluded from operational carbon emissions because they are considered to be negligible.Maintenance costs are the total costs and carbon emissions associatedwith labour, material and other resources used to ensure that a constructed asset remains in a state in which it can fulfil its required functions. They include corrective and preventative maintenance, such as cleaning, services, repainting, repairing or replacing parts that are not classed as renewal and that the client wishes to be included in the maintenance budget.
Decommissioning | End-of-Life: At the end of an infrastructure asset’s useful life or when it has become obsolete or redundant, owners and operators have opportunities to maximize the utility of its materials, components and equipment as well as the land it occupies. While these actions do not enhance the resilience of the infrastructure itself, they support broader societal well-being and resilience. | End of life costs are the net costs or fees (after deducting the salvagevalue and other income due to disposal) and/or carbon emissions associated with disposal of an asset at the end of its service life, including those resulting from:• disposal inspection (for costs only, not carbon emissions)• decommissioning and decontamination• demolition and reclamation• reinstatement• asset transfer obligations• recycling (for costs only, not carbon emissions, since credits for recycling are consideredto lie outwith the project boundaries)• recovery• disposal of components and materials and• transport and regulatory costs.Costs include all constructors’ site overheads, risk allowances and taxes and levies, though carbon emissions associated with inspection, and taxes and levies are considered negligible and should not be reported.

</div>

We propose updating the [projectStatus](https://standard.open-contracting.org/infrastructure/latest/en/reference/codelists/#projectstatus) codelist as follows:

<div dir="ltr" style="margin-left:0pt;" align="left" id="docs-internal-guid-dec34e2d-7fff-d167-e2c4-1bf657cbd86f">

Code | Title | Description | Change
-- | -- | -- | --
identification | Identification | Identification refers to the decision to develop a project within the budget and programme of a project owner (the public entity responsible for executing the budget). **This stage is also called definition.** | Changed
preparation | Preparation | Preparation covers the feasibility study, environmental and social impact assessment, general scoping of the project, establishing the packaging and procurement strategy, preliminary statutory requirements on environmental and land impacts, and the resulting budget authorization. | No change
implementation | implementation | Implementation covers the procurement and implementation of the group of works or services (such as design and supervision) to be delivered under the project, including works or services performed by the procuring entity. This differs from the definition of 'implementation' in OCDS, which covers the implementation but not the procurement. **This stage is also called construction.** | Changed
completion | Completion | Completion covers the handover of the assets and close-out activities with details of the final scope, cost, and delivery time **before maintenance**. | Changed
maintenance | Maintenance | **Maintenance covers any preventative or corrective maintenance and the day-to-day running of the assets. This stage is also called operation.** | New
decommissioning | Decommissioning | **Decommissioning covers the disposal of the assets at the end of their service life. This stage is also called end of life.** | New
completed | Completed | Close-out activities were completed, and this project is inactive. | No change, but deprecated in favour of ‘maintenance’
decommissioned | Decommissioned | **Decommissioning is complete.** | New
cancelled | Cancelled | This project was cancelled before close-out activities were completed, and is inactive. Cancellation can occur anytime after identification **and before maintenance**. | Changed

</div>

where 'identification', 'preparation' and 'implementation' are unchanged but included in the above table for completeness and to show how all codes will look together.

cc @EvelynDinora

### LINKED ISSUE #394: Update project definition to cover maintenance and decommissioning
OC4IDS defines an infrastructure project as:

> the development of a set of infrastructure assets in a specified location, generally the responsibility of a single procuring entity and budget authority: for example, a highway overpass or a university campus.

I think we'll need to update this to cover operation and decommissioning (see #391).

In particular, we'll need to consider the case where the first OC4IDS data published about a project describes its decommissioning.

cc @evelyndinora

Edit: The definition will need updating in the `description` property of `project-schema.json` and in the ['What is a project?'](https://standard.open-contracting.org/staging/infrastructure/0.9-dev/en/projects/#what-is-a-project) section of the getting started documentation.

### LINKED ISSUE #421: Project preparation period and approval period
## Background

This issue relates to the following CoST IDS elements proposed in the CoST IDS/OC4IDS review:

<details> <summary> Project preparation period (preparation) </summary>

## Project preparation period (preparation)

**Module:** Climate finance
**Indicator:** Accessibility/efficiency

### Disclosure format

> Disclose dates for project preparation [start date, end date] 

### OC4IDS mapping

TBC

</details>

<details> <summary> Project approval period (preparation) </summary>

## Project approval period (preparation)

**Module:** Climate finance
**Indicator:** Accessibility

### Disclosure format

> Disclose dates for project approval [submission date, approval date] 

### OC4IDS mapping

TBC

</details>

@evelyndinora, provided the following clarification of the uses cases and meaning of these fields:

> Climate action is needed urgently. But it takes time to prepare a project. What the SG wants to see is the efficiency (or inefficiency) in terms of project preparation. Please see below: 
> image
> https://www.greenclimate.fund/project/fp209 
>
> This is the date the project was submitted to approval of the Climate Fund and the date the project is approved by the Climate Fund. This is applicable to all available climate funds. Then we can use the start date of a project to measure efficiency in financial flows. 

## Discussion

As discussed in https://github.com/open-contracting/infrastructure/issues/420 and elsewhere, there can be several levels of 'project' and investment at play, e.g.

```mermaid
flowchart TD
  A["Funding source,
  e.g. GCF"]-- "'Parent' investment" -->
  B["Financing party, e.g.
  Development Bank of South Africa"]-- "Investment" -->
  C["Infrastructure
    Project 1"] &  D["Infrastructure
    Project 2"] & E["Infrastructure
    Project 3"]
```

@evelyndinora, @mgraca-prado, my understanding is that the 'project' referred to in the elements proposed in the CoST IDS/OC4IDS review is not the infrastructure project, but the 'parent' investment project (not sure if that is the right term exactly). Is that correct?

### PR DESCRIPTION
**Pull request by duncandewhurst**
**Related issues**

* Closes #394 
* Closes #391 
* Closes #252 
* Closes #421

**Merge checklist**

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [x] `docs/examples/example.json`
  - [x] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [ ] Add an entry to `docs/reference/codelists.md`

### CODE REVIEW

File: docs/cost/ids/sustainability.md [hunk 1]
```diff
@@ -1252,9 +1257,28 @@ Disclose dates for project approval \[submission date, approval date\]
...
 
-```json
+For each date:
 
+- Add a `Milestone` to the `milestones` array and set its:
+  - `.id` incrementally
+  - `.type` to 'financing'
+  - `.status` to 'met'
+- For the submission date, set the milestone's `.title` to "Climate finance submission" and its `.dateMet` to the date of the submission
+- For the approval date, set the milestone's `.title` to "Climate finance approval" and its `.dateMet` to the date of the approval
```
Comment (odscjen) [inline docs/cost/ids/sustainability.md]:
```suggestion
- For the approval date, set the milestone's `.title` to "Climate finance approval" and its `.dateMet` to the date of the approval.
```

File: mapping/sustainability.yaml [hunk 2]
```diff
@@ -685,15 +685,42 @@
...
+        Project level:
+
+        For each date:
+
+        - Add a `Milestone` to the `milestones` array and set its:
+          - `.id` incrementally
+          - `.type` to 'financing'
+          - `.status` to 'met'
+        - For the submission date, set the milestone's `.title` to "Climate finance submission" and its `.dateMet` to the date of the submission
+        - For the approval date, set the milestone's `.title` to "Climate finance approval" and its `.dateMet` to the date of the approval
```
Comment (odscjen) [inline mapping/sustainability.yaml]:
```suggestion
        - For the approval date, set the milestone's `.title` to "Climate finance approval" and its `.dateMet` to the date of the approval.
```

**Review by odscjen — CHANGES_REQUESTED** [review]
added a couple of full stops, otherwise looks good to me and ready to be passed to James for his review once the full stops have been added

**Review by odscjen — APPROVED** [review]

File: docs/reference/changelog.md [hunk 3]
```diff
@@ -19,27 +19,41 @@
...
-* [#427](https://github.com/open-contracting/infrastructure/pull/427) - add `Organization.beneficialOwners`
+* Add fields:
+  * [#367](https://github.com/open-contracting/infrastructure/pull/367) - `BudgetBreakdown.approvalDate`
+  * [#379](https://github.com/open-contracting/infrastructure/pull/379) - `.contractingProcesses.summary.tender.datePublished`
+  * [#390](https://github.com/open-contracting/infrastructure/pull/390) - `.identifiers`
+  * [#426](https://github.com/open-contracting/infrastructure/pull/426):
+    * `.transactions`
+    * `.milestones`
+    * `ContractingProcesses.milestones`
+  * [#427](https://github.com/open-contracting/infrastructure/pull/427) - `Organization.beneficialOwners`
```
Comment (jpmckinney) [inline docs/reference/changelog.md]: Use schema order and avoid references to definitions, to make it easier on users. (It's only necessary to refer to definitions when that definition is reused.)

```suggestion
  * [#390](https://github.com/open-contracting/infrastructure/pull/390) - `identifiers`
  * [#427](https://github.com/open-contracting/infrastructure/pull/427) - `parties.beneficialOwners`
  * [#426](https://github.com/open-contracting/infrastructure/pull/426) - `transactions`
  * [#426](https://github.com/open-contracting/infrastructure/pull/426) - `milestones`
  * [#379](https://github.com/open-contracting/infrastructure/pull/379) - `contractingProcesses.summary.tender.datePublished`
  * [#426](https://github.com/open-contracting/infrastructure/pull/426) - `contractingProcesses.summary.milestones`
```

File: docs/reference/changelog.md [hunk 4]
```diff
@@ -19,27 +19,41 @@
...
   * [#369](https://github.com/open-contracting/infrastructure/pull/369) - classificationScheme
   * [#427](https://github.com/open-contracting/infrastructure/pull/427) - country
   * [#426](https://github.com/open-contracting/infrastructure/pull/426) - milestoneType
   * [#426](https://github.com/open-contracting/infrastructure/pull/426) - milestoneStatus
   * [#426](https://github.com/open-contracting/infrastructure/pull/426) - milestoneCode
+* [#432](https://github.com/open-contracting/infrastructure/pull/432):
+  * Add codes to projectStatus codelist:
+    * 'maintenance' 
+    * 'decommissioning'
+    * 'decommissioned;
```
Comment (jpmckinney) [inline docs/reference/changelog.md]:
```suggestion
    * 'decommissioned'
```

File: docs/reference/changelog.md [hunk 5]
```diff
@@ -19,27 +19,41 @@
...
+  * Deprecate 'completed' in favour of 'maintenance'
```
Comment (jpmckinney) [inline docs/reference/changelog.md]:
```suggestion
  * Deprecate 'completed' in favor of 'maintenance' in the projectStatus codelist
```

File: schema/project-level/codelists/projectStatus.csv [hunk 6]
```diff
@@ -1,7 +1,10 @@
...
-implementation,implementation,"Implementation covers the procurement and implementation of the group of works or services (such as design and supervision) to be delivered under the project, including works or services performed by the procuring entity. This differs from the definition of 'implementation' in OCDS, which covers the implementation but not the procurement."
-completion,Completion,"Completion covers the handover of the assets and close-out activities with details of the final scope, cost, and delivery time."
-completed,Completed,"Close-out activities were completed, and this project is inactive."
-cancelled,Cancelled,"This project was cancelled before close-out activities were completed, and is inactive. Cancellation can occur anytime after identification."
+Code,Title,Description,Deprecated,Deprecation note
+identification,Identification,Identification refers to the decision to develop a project within the budget and programme of a project owner (the public entity responsible for executing the budget). This stage is also called definition.,,
```
Comment (jpmckinney) [inline schema/project-level/codelists/projectStatus.csv]:
```suggestion
identification,Identification,Identification refers to the decision to develop a project within the budget and program of a project owner (the public entity responsible for executing the budget). This stage is also called definition.,,
```

File: schema/project-level/codelists/projectStatus.csv [hunk 7]
```diff
@@ -1,7 +1,10 @@
...
+preparation,Preparation,"Preparation covers the feasibility study, environmental and social impact assessment, general scoping of the project, establishing the packaging and procurement strategy, preliminary statutory requirements on environmental and land impacts, and the resulting budget authorization.",,
+implementation,implementation,"Implementation covers the procurement and implementation of the group of works or services (such as design and supervision) to be delivered under the project, including works or services performed by the procuring entity. This differs from the definition of 'implementation' in OCDS, which covers the implementation but not the procurement. This stage is also called construction.",,
```
Comment (jpmckinney) [inline schema/project-level/codelists/projectStatus.csv]:
```suggestion
implementation,implementation,"Implementation covers the procurement and implementation of the group of works or services (such as design and supervision) to be delivered under the project, including works or services performed by the procuring entity. This differs from the definition of implementation in OCDS, which covers the implementation but not the procurement. This stage is also called construction.",,
```

File: schema/project-level/project-schema.json [hunk 8]
```diff
@@ -62,6 +65,36 @@
       "description": "The period over which this project is planned to run. This may be updated during the preparation phase as information becomes available to more accurately specify anticipated start and end dates, but should not be updated during the implementation or completion phases. The planned completion date of the project should be provided in `period.endDate`, which can be compared with the actual completion date in `completion.endDate`",
       "$ref": "#/definitions/Period"
     },
+    "identificationPeriod": {
+      "title": "Identification period",
+      "description": "Identification refers to the decision to develop a project within the budget and programme of a project owner (the public entity responsible for executing the budget). This stage is also called definition.",
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]: I find it a bit odd that the codelist descriptions are reused here, and that there's no reference to "period".

I suggest using the format: "The period over which the {stage-name} stage is planned to run." (following the pattern set by `period`). The meaning of the stages is deferred to the projectStatus codelist.

Comment (duncandewhurst) [reply schema/project-level/project-schema.json]: Sounds good - that'll be easier to maintain, too. Done in https://github.com/open-contracting/infrastructure/pull/432/commits/390d4b022836367b24bf997288e0d03e26a8e6c1

**Comment by duncandewhurst** [comment]
I tidied up the changelog whilst resolving merge conflicts. Feel free to make further edits.

File: docs/reference/changelog.md [hunk 9]
```diff
@@ -19,35 +19,50 @@
...
+    * `completionPeriod`
+    * `maintenancePeriod`
+    * `decommissioningPeriod`
+  * [#390](https://github.com/open-contracting/infrastructure/pull/390) - `identifiers`.
+  * [#426](https://github.com/open-contracting/infrastructure/pull/426):
+    * `.transactions`
+    * `.milestones`
+    * `ContractingProcesses.milestones`
+  * [#431](https://github.com/open-contracting/infrastructure/pull/431) - `environment.goals`.
+  * [#427](https://github.com/open-contracting/infrastructure/pull/427) - `Organization.beneficialOwners`
```
Comment (jpmckinney) [inline docs/reference/changelog.md]: These were duplicated during the merge, I think.

```suggestion
  * [#431](https://github.com/open-contracting/infrastructure/pull/431) - `environment.goals`
```

**Review by jpmckinney — APPROVED** [review]
Just one cleanup, then feel free to merge.

**Pull request merged**